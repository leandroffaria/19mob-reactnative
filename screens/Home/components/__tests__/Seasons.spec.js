import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import Seasons from '../Seasons';

const mockFn = jest.fn();
const mockYears = [
    { season: 2020 },
    { season: 2019 },
    { season: 2018 },
];

const element = render(<Seasons handlerSeason={ mockFn } data={ mockYears } />);

describe('Home -> components -> Seasons', () => {
    it('should be simulate click', () => {
        const { getByTestId } = element;

        const button = getByTestId('year-2020');
        fireEvent.press(button);
    });

    it('should be render snapshot', () => {
        const { container } = element;
        expect(container).toMatchSnapshot();
    });
});