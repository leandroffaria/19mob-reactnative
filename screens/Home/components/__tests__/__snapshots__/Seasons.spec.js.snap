// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home -> components -> Seasons should be render snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Button
    style={
      Object {
        "backgroundColor": "#3483fa",
        "marginBottom": 10,
      }
    }
    testID="year-2020"
    title={2020}
  />
  <Button
    style={
      Object {
        "backgroundColor": "#3483fa",
        "marginBottom": 10,
      }
    }
    testID="year-2019"
    title={2019}
  />
  <Button
    style={
      Object {
        "backgroundColor": "#3483fa",
        "marginBottom": 10,
      }
    }
    testID="year-2018"
    title={2018}
  />
</View>
`;
